### ISSUE TITLE
Leverage git deps for resolving binaries?

### ISSUE THREAD

**Issue by borkdude** [babashka/babashka.pods#19]
- If checking in binaries into git wasn't an anti-pattern, I think Clojure git deps could be used to implement an ecosystem of binaries more or less like brew, but using deps.edn. The classpath can then be used as the binary path, a bit similar to a nix env. I would have a purpose for this: babashka pods.

- Babashka could automatically search the classpath for binaries and invoke them on demand. Or we could have a function like `io/resource` which resolves a URL, i.e. a file on the filesystem. 

- The git deps stuff could be made to work with GraalVM as well I think, so everything could be very fast after deps have been downloaded (but even without GraalVM, it's already fast due to classpath caching).

- Maybe instead of checking in the binaries themselves, there could be links to releases in the source, with checksums, etc in an edn file (a bit like: https://github.com/littleli/scoop-clojure/blob/master/deps.clj.json). But most of the gnarly dependency stuff could be handled by tda using normal `deps.edn`. The `io/resource`-like function could then process that EDN file which contains the links to download the binary for the current OS and architecture and store it in a predictable/cache-able location. Maybe something like `(resolve-binary 'clj-kondo/clj-kondo)` which searches for `clj-kondo/clj-kondo.$foobar.edn` on the classpath (where `$foobar` is the name of this new mechanism/library/higher level package manager) and pull the binar(y)(ies). 

So in summary, we could have:
``` clojure
(pods/load-pod (resolve-binary 'my.cool.pod))
``` 
where `(resolve-binary 'my.cool.pod)` searches for `my/cool/pod.$foobar.edn` on the classpath and pulls the correct binary for your OS to `.$foobar/my/cool/pod/SHA/<the-binary>` if it doesn't exist there yet and returns that file path.

**Comment by borkdude** [comment]
Initial implementation of this idea: https://github.com/borkdude/cpm
